src/targets/__tests__/symbolCollector.test.ts

```diff
@@ -0,0 +1,138 @@
+import { withTempDir } from '../../utils/files';
+import { NoneArtifactProvider } from '../../artifact_providers/none';
+import { checkExecutableIsPresent, spawnProcess } from '../../utils/system';
+import { SymbolCollector, SYM_COLLECTOR_BIN_NAME } from '../symbolCollector';
+
+jest.mock('../../utils/files');
+jest.mock('../../utils/system');
+jest.mock('fs', () => {
+  const original = jest.requireActual('fs');
+  return {
+    ...original,
+    promises: {
+      mkdir: jest.fn(() => {
+        /** do nothing */
+      }),
+    },
+  };
+});
+
+const customConfig = {
+  batchType: 'batchType',
+  bundleIdPrefix: 'bundleIdPrefix-',
+};
+
+function getSymbolCollectorInstance(
+  config: Record<string, unknown> = { ['testKey']: 'testVal' }
```

Comment:
We don't need to use an explicit string key.

```suggestion
  config: Record<string, unknown> = { testKey: 'testVal' }
```